R/woods_plot.R

```diff
@@ -69,86 +71,185 @@ woods_plot <- function(data, fold_change, start_position, end_position, protein_
-      fill = {{ colouring }}
-    ),
-    col = "black",
-    size = 0.7
-    ) +
-    ggplot2::geom_hline(
-      yintercept = -{{ fold_change_cutoff }},
-      col = "blue",
-      alpha = .8,
-      size = 0.7
-    ) +
-    ggplot2::geom_hline(
-      yintercept = {{ fold_change_cutoff }},
-      col = "blue",
-      alpha = .8,
-      size = 0.7
-    ) +
-    ggplot2::ylim(min(-2.5, dplyr::pull(data, {{ fold_change }})) - 0.5, max(2.5, dplyr::pull(data, {{ fold_change }})) + 0.5) +
-    ggplot2::scale_x_continuous(limits = NULL, expand = c(0, 0)) +
-    ggplot2::labs(
-      x = "Protein Sequence",
-      y = "log2(fold change)",
-      title = {
-        if (!missing(protein_id)) unique(dplyr::pull(data, {{ protein_id }}))
-      }
-    ) +
-    {
-      if (!missing(facet)) ggplot2::facet_wrap(rlang::new_formula(NULL, rlang::enquo(facet)), scales = "free", ncol = 4)
-    } +
-    ggplot2::guides(size = FALSE) +
-    {
-      if (!missing(colouring) && !is.numeric(dplyr::pull(data, {{ colouring }}))) ggplot2::scale_fill_manual(values = protti_colours)
-    } +
-    {
-      if (!missing(colouring) && is.numeric(dplyr::pull(data, {{ colouring }}))) ggplot2::scale_fill_gradient(low = protti_colours[1], high = protti_colours[2])
-    } +
-    ggplot2::theme_bw() +
-    ggplot2::theme(
-      plot.title = ggplot2::element_text(
-        size = 20
+  if(!missing(highlight)) {
+    if(!all(is.logical(dplyr::pull(data, {{ highlight }} )))){
+      warning(paste("Please only provide logicals (i.e. TRUE or FALSE) in the 'highlight' column."))
+    }else{
+    data %>%
+      ggplot2::ggplot() +
+      ggplot2::geom_rect(ggplot2::aes(
+        xmin = 0,
+        xmax = {{ protein_length }},
+        ymin = -0.01,
+        ymax = 0.01
       ),
-      axis.text.x = ggplot2::element_text(
-        size = 15
+      fill = "black"
+      ) +
+      ggplot2::geom_rect(ggplot2::aes(
+        xmin = {{ start_position }},
+        xmax = {{ end_position }},
+        ymin = {{ fold_change }} - 0.2,
+        ymax = {{ fold_change }} + 0.2,
+        fill = {{ colouring }}
       ),
-      axis.text.y = ggplot2::element_text(
-        size = 15
-      ),
-      axis.title.y = ggplot2::element_text(
-        size = 15
-      ),
-      axis.title.x = ggplot2::element_text(
-        size = 15
-      ),
-      legend.title = ggplot2::element_text(
-        size = 15
-      ),
-      legend.text = ggplot2::element_text(
-        size = 15
-      ),
-      strip.text.x = ggplot2::element_text(
-        size = 15
+      col = "black",
+      size = 0.7,
+      alpha = 0.8
+      ) +
+      ggplot2::geom_point(data = dplyr::filter(data, {{ highlight }} == TRUE),
+                          ggplot2::aes(
+                            x = (( {{ start_position }} + {{ end_position }} ) /2),
+                            y = ( {{ fold_change }} - 0.3)), 
+                          shape = 8, 
+                          col = "black", 
+                          size = 3) +
```

Comment:
I would not copy paste the whole plot so that you have the whole code twice with just this change. It is still possible to only have it once in there. Just wrap the whole geom_point into an if statement similar to how it is done with the `colouring` argument.
```suggestion
      {if (!missing(highlight){ ggplot2::geom_point(data = dplyr::filter(data, {{ highlight }} == TRUE),
                          ggplot2::aes(
                            x = (( {{ start_position }} + {{ end_position }} ) /2),
                            y = ( {{ fold_change }} - 0.3)), 
                          shape = 8, 
                          col = "black", 
                          size = 3) }} +
```